### PR TITLE
Add mask reduction opcodes

### PR DESCRIPTION
**Pull request by gita-omr**
- mTrueCount, mFirstTrue, mLastTrue, mToLong

### CODE REVIEW

**Comment by gita-omr** [comment]
Jenkins build all

File: compiler/il/VectorOperations.enum [hunk 1]
```diff
@@ -146,6 +146,70 @@ VECTOR_OPERATION_MACRO(\
...
+   /* .typeProperties          = */ TR::NoType, \
+   /* .childProperties         = */ ONE_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    mask last true */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mToLong, \
```
Comment (BradleyWood) [inline compiler/il/VectorOperations.enum:198]: I think the name of this opcode will cause confusion with m2l opcode.

Comment (gita-omr) [reply compiler/il/VectorOperations.enum]: Renamed to mToLongBits

File: compiler/arm/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -1448,6 +1448,18 @@ OMR::ARM::TreeEvaluator::mstoreiEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
+   {
+   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   }
+
+TR::Register*
+OMR::ARM::TreeEvaluator::mLongBitsToMaskEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   }
+
```
Comment (knn-k) [inline compiler/arm/codegen/OMRTreeEvaluator.cpp:1462]: It seems the definitions of the following evaluators are missing for 32 bit ARM.
mTrueCountEvaluator, mFirstTrueEvaluator, mLastTrueEvaluator

Comment (knn-k) [reply compiler/arm/codegen/OMRTreeEvaluator.cpp:1462]: Have you updated your commit?

Comment (knn-k) [reply compiler/arm/codegen/OMRTreeEvaluator.cpp:1462]: The following evaluators are still missing for 32 bit ARM.
mTrueCountEvaluator, mFirstTrueEvaluator, mLastTrueEvaluator

Comment (gita-omr) [reply compiler/arm/codegen/OMRTreeEvaluator.cpp:1480]: You are right! Somehow missed again. I think all should be there now. Sorry about that.

**Comment by knn-k** [comment]
Jenkins build all,amac

**Comment by knn-k** [comment]
There is a compiler error with the RISC-V build in https://ci.eclipse.org/omr/job/PullRequest-linux_riscv64_cross/913/:

```
[2022-11-08T02:36:10.981Z] /home/jenkins/workspace/Build/compiler/riscv/codegen/OMRTreeEvaluator.cpp:1329:1: error: no declaration matches ‘TR::Register* OMR::RV::TreeEvaluator::mLongBitsTomaskEvaluator(TR::Node*, TR::CodeGenerator*)’
[2022-11-08T02:36:10.981Z]  OMR::RV::TreeEvaluator::mLongBitsTomaskEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```

The function name must be `mLongBitsToMaskEvaluator` (capital "M" for "Mask") there.

**Comment by gita-omr** [comment]
> There is a compiler error with the RISC-V build in https://ci.eclipse.org/omr/job/PullRequest-linux_riscv64_cross/913/:
> 
> ```
> [2022-11-08T02:36:10.981Z] /home/jenkins/workspace/Build/compiler/riscv/codegen/OMRTreeEvaluator.cpp:1329:1: error: no declaration matches ‘TR::Register* OMR::RV::TreeEvaluator::mLongBitsTomaskEvaluator(TR::Node*, TR::CodeGenerator*)’
> [2022-11-08T02:36:10.981Z]  OMR::RV::TreeEvaluator::mLongBitsTomaskEvaluator(TR::Node *node, TR::CodeGenerator *cg)
> ```
> 
> The function name must be `mLongBitsToMaskEvaluator` (capital "M" for "Mask") there.

Sorry about that. Just fixed.

**Comment by knn-k** [comment]
Jenkins build all

**Comment by knn-k** [comment]
plinux failed with a known problem with `free(): invalid next size (normal)` in `Special/PPCDirectEncodingTest`.  See #6571.

**Review by knn-k — APPROVED** [review]

**Comment by knn-k** [comment]
I am going to merge this PR tomorrow if there is no other comment.

**Pull request merged**